fix: pydantic error in accent_marker and imporve logging

minto1226 opened the pull request:
<!-- 請先閱讀我們手冊的 [How to PR](https://hackmd.io/@sessatakuma/pr_template#How-to-PR) 章節 -->
<!-- 記得修改PR標題成 conventional commit 之標題格式  -->

### 目的
修改`furigana_results`至正確的datatype，以及修改print成logger

### 方法／實作說明
1. 將`mark_furigana()`改為直接return `Response` object.
2. 將內部 print 更換成 logger
3. Error時result會return None而不是空list

### 關聯 Issue
fix #36

minto1226 committed:
chore: find bug

fix: fix reading from yahoo_subword

heap: adjustment yahoo_subword's type hint

chore: change furigana_results's data type

minto1226 committed:
chore: change print to logger

minto1226 committed:
chore: change furigana_results's data type

heat: fix httpx problem

chore: fix ruff problem

chore: fix ruff problem

chore: fix ruff problem

wade00754 committed:
style(accent_marker): format with ruff

wade00754 committed:
chore: configure logging in main.py

wade00754 committed:
fix: return None instead of empty list for error responses

wade00754 committed:
refactor: update response models to use union types for optional fields

wade00754 committed:
fix: specify type for yahoo_subword in mark_accent function

wade00754 reviewed (dismissed):
Approved. But please take the recommendations into consideration.

minto1226 committed:
chore: seperate mark_furigana to handler and service

chore: change print to logger

fix: fix ruff problem

fix: fix ruff problem

fix: fix ruff problem

minto1226 committed:
chore: change print to logger

chore: change print to logger

minto1226 commented on api/accent_marker.py:
```diff
@@ -250,45 +259,55 @@ async def mark_accent(
         query_text = neologdn.normalize(request.text, tilde="normalize")
+
+        # 1. 呼叫 Yahoo Furigana
         furigana_response = await mark_furigana(Request(text=query_text), client)
```
Resolved.

wade00754 commented on api/furigana_marker.py:
```diff
@@ -20,7 +20,7 @@
 
 
-class Request(BaseModel):
+class RequestBody(BaseModel):
```
I think there is no need to rename this.

wade00754 commented on api/furigana_marker.py:
```diff
@@ -161,3 +160,10 @@ async def mark_furigana(
+    request: RequestBody,
+    client: httpx.AsyncClient = Depends(get_http_client),
+) -> dict[str, Any]:
+    return await mark_furigana_service(request.text, client)
```
My recommendation is that `mark_furigana_service()` should return a `Response` object instead of its dictionary representation (`model_dump()`) so that you can use the `Response` object directly, which improves the type safety.

wade00754 reviewed (changes requested):
Please take a look at the comments to do futher changes.

minto1226 committed:
feat: revise return dict to return class

minto1226 committed:
chore: update docstring and change post position

chore:revise

wade00754 committed:
refactor: return response model directly in furigana_marker

wade00754 committed:
refactor(accent_marker): return response model directly

wade00754 committed:
refactor(accent_marker): simplify furigana results handling and logging

wade00754 committed:
fix(accent_marker): correct string formatting in debug log message

wade00754 committed:
feat: update logger name for accent_marker and improve log format

wade00754 reviewed (approved):
I have found that FastAPI can use the Pydantic model as respone so that we can return it directly.
